drivers/counter/counter_mcux_lptmr.c

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (c) 2020 Vestas Wind Systems A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/counter.h>
+#include <fsl_lptmr.h>
+
+struct mcux_lptmr_config {
+	struct counter_config_info info;
+	LPTMR_Type *base;
+	lptmr_prescaler_clock_select_t clk_source;
+	lptmr_prescaler_glitch_value_t prescaler_glitch;
+	bool bypass_prescaler_glitch;
+	lptmr_timer_mode_t mode;
+	lptmr_pin_select_t pin;
+	lptmr_pin_polarity_t polarity;
+	void (*irq_config_func)(struct device *dev);
+};
+
+struct mcux_lptmr_data {
+	counter_top_callback_t top_callback;
+	void *top_user_data;
+};
+
+static int mcux_lptmr_start(struct device *dev)
+{
+	const struct mcux_lptmr_config *config = dev->config->config_info;
+
+	LPTMR_EnableInterrupts(config->base,
+			       kLPTMR_TimerInterruptEnable);
+	LPTMR_StartTimer(config->base);
+
+	return 0;
+}
+
+static int mcux_lptmr_stop(struct device *dev)
+{
+	const struct mcux_lptmr_config *config = dev->config->config_info;
+
+	LPTMR_DisableInterrupts(config->base,
+				kLPTMR_TimerInterruptEnable);
+	LPTMR_StopTimer(config->base);
+
+	return 0;
+}
+
+static int mcux_lptmr_get_value(struct device *dev, u32_t *ticks)
+{
+	const struct mcux_lptmr_config *config = dev->config->config_info;
+
+	*ticks = LPTMR_GetCurrentTimerCount(config->base);
+
+	return 0;
+}
+
+static int mcux_lptmr_set_top_value(struct device *dev,
+				  const struct counter_top_cfg *cfg)
+{
+	const struct mcux_lptmr_config *config = dev->config->config_info;
+	struct mcux_lptmr_data *data = dev->driver_data;
+
+	if (cfg->ticks == 0) {
+		return -EINVAL;
+	}
+
+	data->top_callback = cfg->callback;
+	data->top_user_data = cfg->user_data;
+
+	if (config->base->CSR & LPTMR_CSR_TEN_MASK) {
+		/* Timer already enabled, check flags before resetting */
+		if (cfg->flags & COUNTER_TOP_CFG_DONT_RESET) {
+			return -ENOTSUP;
+		}
+		LPTMR_StopTimer(config->base);
+		LPTMR_SetTimerPeriod(config->base, cfg->ticks);
+		LPTMR_StartTimer(config->base);
+	} else {
+		LPTMR_SetTimerPeriod(config->base, cfg->ticks);
+	}
+
+	return 0;
+}
+
+static u32_t mcux_lptmr_get_pending_int(struct device *dev)
+{
+	const struct mcux_lptmr_config *config = dev->config->config_info;
+	u32_t flags;
+
+	flags = LPTMR_GetStatusFlags(config->base);
+
+	return (flags & LPTMR_CSR_TCF_MASK & LPTMR_CSR_TIE_MASK);
```

Comment:
This logic isn't quite right to check for both bits. It should be something like:
```c
u32_t mask = LPTMR_CSR_TCF_MASK | LPTMR_CSR_TIE_MASK;
return ((flags & mask) == mask);
```

Reply:
Ouch, thanks for catching this - and thank you for reviewing.
Fixed.